app/models/miq_queue.rb

```diff
@@ -364,6 +364,7 @@ def deliver(requester = nil)
       _log.error("#{MiqQueue.format_short_log_msg(self)}, #{message}")
       status = STATUS_EXPIRED
     rescue Exception => error
+      raise if error.class.name == "IRB::Abort"
```

Comment:
Scary thought but maybe we should not rescue Exception?

Turns out, I added the rescue Exception...  😢🏃 

```
commit 49e66c99444aff0042648a83510f82305156369d
Author: Joe Rafaniello <jrafanie@redhat.com>
Date:   Tue Mar 3 17:01:02 2009 +0000

    BugzID: 4942

    Workers were not catching all Exceptions in deliver which caused them to exit their work loop and eventually be killed 10-15 minutes later due to it not heartbeating.

    git-svn-id: http://miq-ubuntusub.manageiq.com/svn/svnrepos/Manageiq/trunk@12826 3c68ef56-dcc3-11dc-9475-a42b84ecc76f

diff --git a/vmdb/app/models/miq_queue.rb b/vmdb/app/models/miq_queue.rb
index 7adb4ed..58eb2df 100644
--- a/vmdb/app/models/miq_queue.rb
+++ b/vmdb/app/models/miq_queue.rb
@@ -206,7 +206,7 @@ class MiqQueue < ActiveRecord::Base
         $log.error("MIQ(MiqQueue.deliver)  (PID: #{Process.pid}) Message id: [#{self.id}], Ident: [#{self.queue_name}], Zone: [#{self.zone}], Role: [#{self.role}], Server: [#{self.server_guid}], Target id: [#{target_id}], Task id: [#{self.task_id}], Command: #{self.cl
         status = "timeout"
       end
-    rescue => error
+    rescue Exception => error^M
       $log.error("MIQ(MiqQueue.deliver)  (PID: #{Process.pid}) Message id: [#{self.id}], Zone: [#{self.zone}], Role: [#{self.role}], Server: [#{self.server_guid}], Ident: [#{self.queue_name}], Target id: [#{target_id}], Task id: [#{self.task_id}], Command: #{self.clas
       $log.log_backtrace(error)
       #puts "MIQ(MiqQueue.deliver) \"#{error}\", Id: #{self.id}, method_name: #{self.method_name}"
```

From BZ: "scheduled host datastorage analysis is toxic to generic workers"

"Our workers were running commands and only rescuing StandardError.  Since SyntaxError is not a StandardError, the code raised the error and no one rescued it.  This caused the worker to sit there until 10-15 when the worker is killed.  We now rescue Exception, the base class of all exceptions."

So, what I should have done back then was be less defensive and fix the issue:
`rescue StandardError, SyntaxError => error`